### PR TITLE
[WIP] Clean repository, Windows support, split one.js in modules, speed up execution with lazy loading, add some features

### PR DESCRIPTION
**Pull request by Alex-D**
**This is a WIP**, I've made this PR to show you what I've done and what I do.

Readable diff: https://github.com/Alex-D/one-gulp/compare/feb5781...master
# What I've done
## Clean and Windows support
- remove node_modules folder: this folder contains compiled packages, so on others system that "UNIX" x64 (I've got some bugs on Windows for instance) and speed up local git (especially on zsh)
- remove bower_components from example folder : speed up clone of one-gulp for builds in non one-gulp develop environnement (in CI, prod, or in dev on anothers projects)
- update some dependencies to get some bugfix (especially sass to avoid some segfaults)
## Improve speed
- move require in method to load only if needed. If I want graph, I don't need babel, autoprefixr or whatever, I just need viz, and some dependencies
- remove babel/register, using harmonize insteed: need node 0.12+ but speed up launch of any task (wins 2+ secs on start on SSD, definitly more on HDD)
- added gulp-cached and gulp-remember on each module improve watch on a big project from 5 - 10 seconds (rebuild the project using all transforms) to less than 1 second by skiping in memory transformed files
## Code clean
- split all transforms in modules by programming language / feature with a list of modules to load (not finished yet)
- remove some unsupported features in one-gulp core code because of use of harmonize
- factorize some code
## New features
- JSON module: source and minify
- add counters and transparency (from @hsablonniere work on another branch of this repository)
- create SVG core module: source, minify, inject
- add `one.insert(...).before(...)` and `one.insert(...).after(...)` to slim gulpfile.js user code
- add `one.clone(<function>, <newName>)` method for identical transforms used in prod and dev and ...
## Other
- allow options with more that 1 deep level to support options with multiple level in modules
- put all dependencies options used avaible and customizable in one-gulp options (for sass, html-minify, ...)
# Things to do to end this PR in my opinion
- properly manage one-gulp core modules (only link loded modules)
- add ability to inject modules in one-gulp to make it extensible by adding other packages (I imagine one-gulp-angular, one-gulp-browserify, ...)
# Ideas to continue on this path

Make package.json template to dynamically manage list of dependencies needed by modules loaded in user gulpfile.js options.modules array to improve installation time by "removing" useless dependencies in his case.

---

I hope you watch what I do and it will advance one-gulp :)

### CODE REVIEW

**Comment by hsablonniere** [comment]
TL;DR (yet)
